packages/block-library/src/embed/embed-preview.js

```diff
@@ -18,150 +18,131 @@ import {
 	BlockIcon,
 	__experimentalGetElementClassName,
 } from '@wordpress/block-editor';
-import { Component } from '@wordpress/element';
 import { createBlock, getDefaultBlockName } from '@wordpress/blocks';
+import { useState, useEffect } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import WpEmbedPreview from './wp-embed-preview';
 
-class EmbedPreview extends Component {
-	constructor() {
-		super( ...arguments );
-		this.hideOverlay = this.hideOverlay.bind( this );
-		this.state = {
-			interactive: false,
-		};
-	}
+export default function EmbedPreview( {
+	preview,
+	previewable,
+	url,
+	type,
+	caption,
+	onCaptionChange,
+	isSelected,
+	className,
+	icon,
+	label,
+	insertBlocksAfter,
+} ) {
+	const [ interactive, setInteractive ] = useState( false );
 
-	static getDerivedStateFromProps( nextProps, state ) {
-		if ( ! nextProps.isSelected && state.interactive ) {
-			// We only want to change this when the block is not selected, because changing it when
-			// the block becomes selected makes the overlap disappear too early. Hiding the overlay
-			// happens on mouseup when the overlay is clicked.
-			return { interactive: false };
+	useEffect( () => {
+		if ( isSelected && interactive ) {
```

Comment:
We should also add the `interactive` dep.
```suggestion
		if ( ! isSelected && interactive ) {
```

Reply:
Also consider just [calling `setState` during rendering](https://react.dev/reference/react/Component#static-getderivedstatefromprops) to replace `getDerivedStateFromProps`. Something like:

```js
if ( ! isSelected && interactive ) {
  setInteractive( false );
}
```

Reply:
Thanks for the suggestions and the missed negative condition. I updated it in this commit: https://github.com/WordPress/gutenberg/pull/51325/commits/f4349e489e7d6f7a91caf592218df87af57e38ac
I also rebased the PR.